# MPI_Comm is a typedef struct ompi_communicator_t *MPI_Comm.
# Csys2blacs_handle is called only here and returns a hardcoded zero
# this patch does not alter the behavior 
# Martin Golasowski, martin.golasowski@vsb.cz, IT4Innovations 2017
diff -ru qball_r140/trunk/src/Context.C qball_r140_working/trunk/src/Context.C
--- qball_r140/trunk/src/Context.C	2013-04-18 20:31:13.000000000 +0200
+++ qball_r140_working/trunk/src/Context.C	2017-03-14 19:36:35.998218396 +0100
@@ -613,7 +613,7 @@
   char order = 'C';
   MPI_Comm_size(comm_,&nprocs);
   MPI_Comm_rank(comm_,&mype_);
-  int bhandle = Csys2blacs_handle(comm_);
+  int bhandle = 0;
   ictxt_ = bhandle;
   Cblacs_gridinit( &ictxt_, &order, nprow, npcol );
 
diff -ru qball_r140/trunk/src/Makefile qball_r140_working/trunk/src/Makefile
--- qball_r140/trunk/src/Makefile	2013-07-03 22:10:55.000000000 +0200
+++ qball_r140_working/trunk/src/Makefile	2017-03-14 19:15:26.870687572 +0100
@@ -7,7 +7,19 @@
 
 # If defined, use architecture file set with ARCH variable
 ARCH ?= $(ARCHGUESS)
-include $(ARCH).mk
+PLT=LINUX_X86_64_GNU
+FFTWDIR=${EBROOTFFTW}
+FFTWLIB=$(FFTWDIR)/lib/libfftw.a
+SCALAPACK_DIR = ${EBROOTSCALAPACK}
+SCALAPACKLIB  = $(SCALAPACK_DIR)/lib/libscalapack.a
+BLASDIR = ${EBROOTOPENBLAS}
+DFLAGS += -DUSE_CSTDIO_LFS -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
+INCLUDE = -I$(FFTWDIR)/include
+CXXFLAGS= -DUSE_MPI -DSCALAPACK -DADD_ -D$(PLT) $(INCLUDE) $(DFLAGS)
+LD = ${CXX}
+LIBPATH = -L$(FFTWDIR) -L$(BLASDIR) -L$(EBROOTOPENMPI) -L$(SCALAPACK_DIR)/lib
+LIBS =  -lscalapack -fopenmp -lopenblas -lgfortran
+LDFLAGS = $(LIBPATH) $(LIBS) 
 #------------------------------------------------------------------------------
 #
 TESTEXECS=testMatrix testFTGrid testGridFunction testBasis \
